Dockerfile.amd64

```diff
@@ -16,52 +16,147 @@ ARG BIRD_IMAGE=calico/bird:latest
+    yum-builddep -y --spec /root/rpmbuild/SPECS/libnftnl.spec && \
+    rpmbuild -bb /root/rpmbuild/SPECS/libnftnl.spec && \
+    # Now install libnftnl and libnftnl-devel
+    rpm -Uv /root/rpmbuild/RPMS/${ARCH}/libnftnl-${LIBNFTNL_VER}.el8.${ARCH}.rpm && \
+    rpm -Uv /root/rpmbuild/RPMS/${ARCH}/libnftnl-devel-${LIBNFTNL_VER}.el8.${ARCH}.rpm && \
+    # Install source RPM for iptables and install its build dependencies.
+    rpm -i ${IPTABLES_SOURCERPM_URL} && \
+    yum-builddep -y --spec /root/rpmbuild/SPECS/iptables.spec
+
+# Patch the iptables build spec so that we keep the legacy iptables binaries.
+RUN sed -i '/drop all legacy tools/,/sbindir.*legacy/d' /root/rpmbuild/SPECS/iptables.spec
+
+# Patch the iptables build spec to drop the renaming of nft binaries. Instead of renaming binaries,
+# we will use alternatives to set the canonical iptables binaries.
+RUN sed -i '/rename nft versions to standard name/,/^done/d' /root/rpmbuild/SPECS/iptables.spec
+
+# Patch the iptables build spec so that legacy and nft iptables binaries are verified to be in the resulting rpm.
+RUN sed -i '/%files$/a \
+\%\{_sbindir\}\/xtables-legacy-multi \n\
+\%\{_sbindir\}\/ip6tables-legacy \n\
+\%\{_sbindir\}\/ip6tables-legacy-restore \n\
+\%\{_sbindir\}\/ip6tables-legacy-save \n\
+\%\{_sbindir\}\/iptables-legacy \n\
+\%\{_sbindir\}\/iptables-legacy-restore \n\
+\%\{_sbindir\}\/iptables-legacy-save \n\
+\%\{_sbindir\}\/ip6tables-nft\n\
+\%\{_sbindir\}\/ip6tables-nft-restore\n\
+\%\{_sbindir\}\/ip6tables-nft-save\n\
+\%\{_sbindir\}\/iptables-nft\n\
+\%\{_sbindir\}\/iptables-nft-restore\n\
+\%\{_sbindir\}\/iptables-nft-save\n\
+' /root/rpmbuild/SPECS/iptables.spec
+
+# Finally rebuild iptables.
+RUN rpmbuild -bb /root/rpmbuild/SPECS/iptables.spec
+
+# runit is not available in ubi or CentOS repos so build it.
+RUN wget -P /tmp http://smarden.org/runit/runit-${RUNIT_VER}.tar.gz && \
+    gunzip /tmp/runit-${RUNIT_VER}.tar.gz && \
+    tar -xpf /tmp/runit-${RUNIT_VER}.tar -C /tmp && \
+    cd /tmp/admin/runit-${RUNIT_VER}/ && \
+    package/install
+
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
+
+ARG GIT_VERSION=unknown
+ARG ARCH=x86_64
+ARG IPTABLES_VER=1.8.2-9
+ARG RUNIT_VER=2.1.2
+
+# Required labels for certification
+LABEL name="Calico node" \
+      vendor="Project Calico" \
+      version=$GIT_VERSION \
+      release="1" \
+      summary="Calico node handles networking and policy for Calico" \
+      description="Calico node handles networking and policy for Calico" \
+      maintainer="laurence@tigera.io"
+
+# Copy in runit binaries
+COPY --from=centos  /tmp/admin/runit-${RUNIT_VER}/command/* /usr/local/bin/
+
+# Copy in our rpms
+COPY --from=centos  /root/rpmbuild/RPMS/${ARCH}/* /tmp/rpms/
+
+COPY centos.repo /etc/yum.repos.d/
+
+# Install everything but conntrack, making sure that we're using only CentOS repos.
+# We're using CentOS for all our packages for consistency and probably better compatibility.
```

Comment:
What do you mean by "probably better compatibility" here?

Reply:
I wanted to convey that there is less chance of package dependency issues if all of our packages were from a single source (CentOS) rather than a mix of CentOS and ubi. I'll reword